tidy3d/plugins/dispersion/web.py

```diff
@@ -0,0 +1,362 @@
+        headers = self._setup_server(url_server)
+
+        resp = requests.post(
+            f"{url_server}/dispersion/fit",
+            headers=headers,
+            data=self.json(),
+            verify=DEFAULT_CONFIG.ssl_verify,
+        )
+
+        try:
+            resp.raise_for_status()
+        except Exception as e:
+            if resp.status_code == ExceptionCodes.GATEWAY_TIMEOUT.value:
+                msg = (
+                    (
+                        "Fitter failed due to timeout. Try to decrease the number of tries or "
+                        "inner iterations, to relax the RMS tolerance, or to use the 'hard' "
+                        "constraint."
+                    )
+                    if self.constraint != "hard"
+                    else (
+                        "Fitter failed due to timeout. Try to decrease the number of tries or "
+                        "inner iterations, or to relax the RMS tolerance."
+                    )
+                )
+                raise Tidy3dError(msg) from e
+
+            raise WebError(
+                "Fitter failed. Try again, tune the parameters, or contact us for more help."
+            ) from e
+
+        run_result = resp.json()
+        best_medium = PoleResidue.parse_raw(run_result["message"])
+        best_rms = float(run_result["rms"])
+
+        if best_rms < self.tolerance_rms:
+            log.info("Found optimal fit with RMS error %.3g", best_rms)
+        else:
+            log.warning(
+                "Unable to fit with RMS error under 'tolerance_rms' of %.3g", self.tolerance_rms
+            )
+            log.info("Returning best fit with RMS error %.3g", best_rms)
+
+        return best_medium, best_rms
+
+
+def run(
+    fitter: DispersionFitter,
+    num_poles: PositiveInt = 1,
+    num_tries: PositiveInt = 50,
+    tolerance_rms: NonNegativeFloat = 1e-2,
+    advanced_param: AdvancedFitterParam = AdvancedFitterParam(),
+) -> Tuple[PoleResidue, float]:
+    """Execute the data fit using the stable fitter in the server.
+
+    Parameters
+    ----------
+    fitter : DispersionFitter
+        Fitter with the data to fit.
+    num_poles : PositiveInt, optional
+        Number of poles in the model.
+    num_tries : PositiveInt, optional
+        Number of optimizations to run with random initial guess.
+    tolerance_rms : NonNegativeFloat, optional
+        RMS error below which the fit is successful and the result is returned.
+    advanced_param : :class:`AdvancedFitterParam`, optional
+        Advanced parameters passed on to the server.
+
+    Returns
+    -------
+    Tuple[:class:`.PoleResidue`, float]
+        Best results of multiple fits: (dispersive medium, RMS error).
+    """
+    task = FitterData.create(fitter, num_poles, num_tries, tolerance_rms, advanced_param)
+    return task.run()
+
+
+class StableDispersionFitter(DispersionFitter):
+    """Deprecated."""
+
+    def __init__(self, *args, **kwargs):
```

Comment:
we can also do this through a simple root_validator

```py
@pd.root_validator()
def _depredate_stable_fitter(cls, values):
        log.warning(
             "'StableDispersionFitter' has been deprecated. Use 'DispersionFitter' with "
             "'tidy3d.plugins.dispersion.web.run' to access the stable fitter from the web server."
     )
    return values

```

I think I'd prefer this to avoid messing with the init method in pydantic

Reply:
Agreed, I don't like playing with `__init__` in those cases either.